### ISSUE TITLE
About the accuracy on SVHN dataset

### ISSUE THREAD

**Issue by IsaccGuang** [snudatalab/KegNet#2]
Thanks for the good work. I try to reproduce the paper results. I have achieved similar accuracy on MNIST dataset but failed on SVHN dataset. Anything is the same but `dataset = 'mnist'` to `dataset = 'svhn'` in `main.py`, which gives accuracy around 52.06%. I want to ask if there is anything else that should be changed to reach the paper accuracy. Thanks again.

**Comment by jaeminyoo** [comment]
@IsaccGuang 
I found the code that we used for the experiments. The exact hyperparameter settings are given as follows. You should change alpha, beta, initial learning rate, and the structures of networks. Please contact us if you cannot reproduce the results with these settings. Thanks!
```
if dataset == 'mnist':
  dec_layers = 1
  lrn_rate = 1e-3
  model = 'lenet5'
  alpha = 1
  beta = 0
elif dataset == 'fashion':
  dec_layers = 3
  lrn_rate = 1e-2
  model = 'resnet14'
  alpha = 1
  beta = 10
elif dataset == 'svhn':
  dec_layers = 3
  lrn_rate = 1e-2
  model = 'resnet14'
  alpha = 1
  beta = 1
```

**Comment by IsaccGuang** [comment]
@JaeminYoo Thanks for replying. I have double checked that the hyperparameter settings you give and those in the code are the same. The following settings are from line 86-105 of `kegnet/generator/train.py`.  
```
if dataset == 'mnist':
    dec_layers = 1
    lrn_rate = 1e-3
    alpha = 1
    beta = 0
elif dataset == 'fashion':
    dec_layers = 3
    lrn_rate = 1e-2
    alpha = 1
    beta = 10
elif dataset == 'svhn':
    dec_layers = 3
    lrn_rate = 1e-2
    alpha = 1
    beta = 1
else:
    dec_layers = 2
    lrn_rate = 1e-4
    alpha = 1
    beta = 0
```
As for `model`, it is set by `cls_network = cls_utils.init_classifier(dataset).to(DEVICE)` in line 107 as the following code shows. 
```
def init_classifier(dataset):
    d = data.to_dataset(dataset)
    if dataset == 'mnist':
        return lenet.LeNet5()
    elif dataset in ('svhn', 'fashion'):
        return resnet.ResNet(d.nc, d.ny)
    else:
        return linear.MLP(d.nx, d.ny)
```
I think the settings are the same so the code will give the same results. I wonder whether there exists anything else that needs to pay attention to.

**Comment by jaeminyoo** [comment]
@IsaccGuang Sorry for the late reply. I have performed more experiments and found that the parameter distribution of a teacher model is a crucial factor that affects the accuracy of distilled classifiers. This is more relevant to how much the parameters are friendly to Tucker decomposition, as the distillation is done after decomposing the weight tensors by Tucker decomposition, rather than the actual accuracy of the teacher model. I think that this property applies to every data-free distillation model that is based on Tucker decomposition, including the baseline approaches that generate data examples from the uniform or Gaussian distributions. 

The reason why the accuracy is not reproduced is because I re-trained all teacher models before releasing the code when improving the code quality. Maybe you can try with different teacher models (with different random seeds, epochs, or other training hyperparameters) to see how the accuracy of distilled classifiers changes.

**Comment by IsaccGuang** [comment]
@JaeminYoo Well, I have already done some other distilling work (like this one) and I think you are right that the hyper-parameter settings of teacher models have a big influence on the distilling results. But it's really hard to say how different teacher models (with different random seeds, epochs, or other training hyperparameters) will affect the final distilling results. And maybe current work cannot provide a stable distilling procedure. 

Now, the only way to reproduce the SVHN results might be using the respect trained teacher model (not the one I used in the repo). Whatever, thanks for your explanation.
